### PR TITLE
EZP-29695: Option "Default content availability" on content types not working

### PR DESCRIPTION
**Pull request by ViniTou**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-29695](https://jira.ez.no/browse/EZP-29695)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `6.x`/`7.x`
| **BC breaks**      | yes
| **Tests pass**     | yes/no
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

**Comment by ViniTou** [comment]
This should be rebased for 5.4 (if ok) or somewhere higher?

**Comment by andrerom** [comment]
6.7, we cherry pick for 5.4

File: eZ/Publish/Core/Repository/ContentService.php [hunk 1]
```diff
@@ -522,7 +522,7 @@ public function createContent(APIContentCreateStruct $contentCreateStruct, array
         }
 
         if ($contentCreateStruct->alwaysAvailable === null) {
-            $contentCreateStruct->alwaysAvailable = false;
+            $contentCreateStruct->alwaysAvailable = $contentCreateStruct->contentType->defaultAlwaysAvailable ?? false;
```
Comment (andrerom) [inline eZ/Publish/Core/Repository/ContentService.php]: `??` is PHP 7 only

Comment (ViniTou) [reply eZ/Publish/Core/Repository/ContentService.php]: It's easy to get use to nice things ;)

**Comment by andrerom** [comment]
You'll need to re-push here. If you push _after_ you change base branch, Travis picks the right one :)

File: eZ/Publish/Core/Repository/ContentService.php (hunk 1 above)
Comment (alongosz) [reply eZ/Publish/Core/Repository/ContentService.php]: Well, but for older versions we still could keep it shorter:
```php
 $contentCreateStruct->contentType->defaultAlwaysAvailable ?: false;
```
:stuck_out_tongue:

**Comment by lserwatka** [comment]
Yes, tomorrow morning.

**Comment by micszo** [comment]
On Studio v1.7.7 language mask seems to be set correctly without the fix. Are you sure base is correct?
I will check on v2.2.3.

**Comment by andrerom** [comment]
Thanks @ViniTou!

And to answer your question @micszo, 1.x UI isn't affected since it uses REST which is afaik correctly using `newContentCreateStruct()` which already has a fix. But any code not using that an initiating the struct manually, is affected by this, hence the backport to strengthen kernel across all branches.

@ViniTou Any idea iof repo-forms could be changed to use the methods to get structs? If so it would avoid future issues like this when we want to fix defaults on structs via these official factory methods.

**Comment by andrerom** [comment]
Ah, I recognize this now, ref:
```
/**
 * @property-read \EzSystems\RepositoryForms\Data\Content\FieldData[] $fieldsData
 */
class ContentCreateData extends ContentCreateStruct implements NewnessCheckable
```

So it's not that simple  then as long as structs in api are not suitable to be used directly in repo forms.

From my side we can beef up API so it fits better for direct usage like this, both in terms of structs, but also to make the mapping to form types "closer to the metal" so to say. I understanding engineering had similar thoughts in mind for 3.x to simplify things?

**Pull request merged**